src/Microsoft.Identity.Web/DownstreamWebApiSupport/IDownstreamWebApi.cs

```diff
@@ -8,101 +8,101 @@
-        /// <returns>The value returned by the downstream web API.</returns>
-        /// <example>
-        /// A list method that returns an IEnumerable&lt;MyItem&gt;&gt;.
-        /// <code>
-        /// public async Task&lt;IEnumerable&lt;MyItem&gt;&gt; GetAsync()
-        /// {
-        ///  return await _downstreamWebApi.CallWebApiForUserAsync&lt;object, IEnumerable&lt;MyItem&gt;&gt;(
-        ///         ServiceName,
-        ///         null,
-        ///         options =>
-        ///         {
-        ///           options.RelativePath = $"api/todolist";
-        ///         });
-        /// }
-        /// </code>
-        ///
-        /// Example of editing.
-        /// <code>
-        /// public async Task&lt;MyItem&gt; EditAsync(MyItem myItem)
-        /// {
-        ///   return await _downstreamWebApi.CallWebApiForUserAsync&lt;MyItem, MyItem&gt;(
-        ///         ServiceName,
-        ///         nyItem,
-        ///         options =>
-        ///         {
-        ///            options.HttpMethod = HttpMethod.Patch;
-        ///            options.RelativePath = $"api/todolist/{myItem.Id}";
-        ///         });
-        /// }
-        /// </code>
-        /// </example>
-        public Task<TOutput?> CallWebApiForUserAsync<TInput, TOutput>(
-            string serviceName,
-            TInput input,
-            Action<DownstreamWebApiOptions>? downstreamWebApiOptionsOverride = null,
-            ClaimsPrincipal? user = null)
-            where TOutput : class;
+		/// <summary>
+		/// Calls a downstream web API consuming JSON with some data and returns data.
+		/// </summary>
+		/// <typeparam name="TInput">Input type.</typeparam>
+		/// <typeparam name="TOutput">Output type.</typeparam>
+		/// <param name="serviceName">Name of the service describing the downstream web API. There can
+		/// be several configuration named sections mapped to a <see cref="DownstreamWebApiOptions"/>,
+		/// each for one downstream web API. You can pass-in null, but in that case <paramref name="downstreamWebApiOptionsOverride"/>
+		/// needs to be set.</param>
+		/// <param name="input">Input parameter to the downstream web API.</param>
+		/// <param name="downstreamWebApiOptionsOverride">Overrides the options proposed in the configuration described
+		/// by <paramref name="serviceName"/>.</param>
+		/// <param name="user">[Optional] Claims representing a user. This is useful in platforms like Blazor
+		/// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
+		/// will find the user from the HttpContext.</param>
+		/// <returns>The value returned by the downstream web API.</returns>
+		/// <example>
+		/// A list method that returns an IEnumerable&lt;MyItem&gt;&gt;.
+		/// <code>
+		/// public async Task&lt;IEnumerable&lt;MyItem&gt;&gt; GetAsync()
+		/// {
+		///  return await _downstreamWebApi.CallWebApiForUserAsync&lt;object, IEnumerable&lt;MyItem&gt;&gt;(
```

Comment:
to illustrate best practices, the task should be returned instead of awaited in the example.